apps/www/registry/new-york/ui/timeline.tsx

```diff
@@ -0,0 +1,207 @@
+import React from "react"
+import { CheckIcon, CircleIcon, Cross1Icon } from "@radix-ui/react-icons"
+import { VariantProps, cva } from "class-variance-authority"
+
+import { cn } from "@/lib/utils"
+
+const timelineVariants = cva("grid", {
+  variants: {
+    positions: {
+      left: "[&>li]:grid-cols-[0_min-content_1fr]",
+      right: "[&>li]:grid-cols-[1fr_min-content]",
+      center: "[&>li]:grid-cols-[1fr_min-content_1fr]",
+    },
+  },
+  defaultVariants: {
+    positions: "left",
+  },
+})
+
+interface TimelineProps
+  extends React.HTMLAttributes<HTMLUListElement>,
+    VariantProps<typeof timelineVariants> {}
+
+const Timeline = React.forwardRef<HTMLUListElement, TimelineProps>(
+  ({ children, className, positions, ...props }, ref) => {
+    return (
+      <ul
+        className={cn(timelineVariants({ positions }), className)}
+        ref={ref}
+        {...props}
+      >
+        {children}
+      </ul>
+    )
+  }
+)
+Timeline.displayName = "Timeline"
+
+const timelineItemVariants = cva("grid items-center gap-x-2", {
+  variants: {
+    status: {
+      done: "text-primary",
+      default: "text-muted-foreground",
+    },
+  },
+  defaultVariants: {
+    status: "default",
+  },
+})
+
+interface TimelineItemProps
+  extends React.HTMLAttributes<HTMLLIElement>,
+    VariantProps<typeof timelineItemVariants> {}
+
+const TimelineItem = React.forwardRef<HTMLLIElement, TimelineItemProps>(
+  ({ className, status, ...props }, ref) => (
+    <li
+      className={cn(timelineItemVariants({ status }), className)}
+      ref={ref}
+      {...props}
+    />
+  )
+)
+TimelineItem.displayName = "TimelineItem"
+
+const timelineDotVariants = cva(
+  "col-start-2 col-end-3 row-start-1 row-end-1 flex size-4 items-center justify-center rounded-full border border-current",
+  {
+    variants: {
+      status: {
+        default: "[&>*]:hidden",
+        current:
+          "[&>*:not(.radix-circle)]:hidden [&>.radix-circle]:bg-current [&>.radix-circle]:fill-current",
+        done: "bg-primary [&>*:not(.radix-check)]:hidden [&>.radix-check]:text-background",
+        error:
+          "border-destructive bg-destructive [&>*:not(.radix-cross)]:hidden [&>.radix-cross]:text-background",
+        custom: "[&>*:not(:nth-child(4))]:hidden [&>*:nth-child(4)]:block",
+      },
+    },
+    defaultVariants: {
+      status: "default",
+    },
+  }
+)
+
+interface TimelineDotProps
+  extends React.HTMLAttributes<HTMLDivElement>,
+    VariantProps<typeof timelineDotVariants> {
+  customIcon?: React.ReactNode
+}
+
+const TimelineDot = React.forwardRef<HTMLDivElement, TimelineDotProps>(
```

Comment:
I suggest having 2 different components eg. TimelineCheckDot and TimelineCrossDot, and removing the status prop